camel-support/src/main/java/io/kaoto/backend/camel/model/deployment/kamelet/serializer/KameletSpecSerializer.java

```diff
@@ -0,0 +1,30 @@
+package io.kaoto.backend.camel.model.deployment.kamelet.serializer;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import io.kaoto.backend.camel.KamelHelper;
+import io.kaoto.backend.camel.model.deployment.kamelet.KameletSpec;
+
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class KameletSpecSerializer extends StdSerializer<KameletSpec> {
```

Comment:
I'm not very sure if I understand the need of such serializer as what it does is to take a `KameletSpec`, translate into a Map and then serializing it back to Json: can the model be fixed instead ?

Reply:
I just did a quick round of replacing things with Jackson, didn't check if with Jackson it could be improved. Going to do another round for this.

Reply:
(This comes from the representer classes that were handling special output cases in snakeyaml)